ios-sample-app/ios-sample-app/AppDelegate.m

```diff
@@ -52,43 +73,91 @@ - (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserAct
     // Starts a new session when the user opens the app using a Singular Link while it was in the background
     SingularConfig *config = [self getConfig];
     config.userActivity = userActivity;
-    
     [Singular start:config];
-    
-    // Request user consent to use the Advertising Identifier (idfa)
-    [self requestTrackingAuthorization];
+        
+    return YES;
+}
 
+- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options{
+    // Starts a new session when the user opens the app using a Non Singular Link, like a traditional App scheme.
+    // This code block will provide the Non Singular link value to the Singular Deeplink Handler
+    SingularConfig *config = [self getConfig];
+    [Singular start:config];
+            
+    // Replace this code with your own logic to handle Non Singular Deeplinks
+    NSLog(@"Handle URL: %@", url);
+    NSString *deeplink = url.absoluteString;
+    if (deeplink == (id)[NSNull null] || deeplink.length == 0 ) deeplink = @"";
```

Comment:
Will  [Utils isEmptyOrNull: deeplink] work?

Reply:
Yes, this also works. Updated to the Utils function.